k3k-kubelet/provider/provider.go

```diff
@@ -665,3 +670,37 @@ func getSecretsAndConfigmaps(pod *corev1.Pod) ([]string, []string) {
 	}
 	return secrets, configMaps
 }
+
+// fetchFieldPathAnnotations will retrieve all annotations created by the pod mutator webhook
+// to assign env fieldpaths to pods
+func (p *Provider) fetchFieldPathAnnotations(pod, tPod *v1.Pod) error {
+	for name, value := range pod.Annotations {
+		var (
+			containerIndex int
+			envName        string
+			err            error
+		)
+		if strings.Contains(name, webhook.FieldpathField) {
+			s := strings.SplitN(name, "_", 3)
+			if len(s) != 3 {
+				continue
+			}
+			containerIndex, err = strconv.Atoi(s[1])
+			if err != nil {
+				return err
+			}
+			envName = s[2]
```

Comment:
I wouldn't try to spread outside this logic, it looks part of the webhook fieldRef stuff. Would it make sense to have inside the `webhook` package two func to handle this?

Something like:

```
func FormatFieldPathAnnotationKey() {
    // formatting logic here
}

func ParseFieldPathAnnotationKey(annotationKey string) (name string, containerIndex int, envName string, error) {
    // parse logic here
}
```

The cleanest way would be have a `FieldPathAnnotationKey` type, but this is up to you. If it's used only here it's fine enough.

Reply:
ah okay, I can add the logic in the webhook package